### PR TITLE
V3

### PR DESCRIPTION
**Pull request by Keats**

### CODE REVIEW

**Comment by Keats** [comment]
@naerbnic does this work for you for the `decode_header`?

@mhseiden @bhendo for the optional `typ`

Any other changes you'd like to see?

File: src/header.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ pub struct Header {
     /// The type of JWS: it can only be "JWT" here
     ///
     /// Defined in [RFC7515#4.1.9](https://tools.ietf.org/html/rfc7515#section-4.1.9).
-    typ: String,
+    typ: Option<String>,
```
Comment (bhendo) [inline src/header.rs]: @Keats Should this attribute be applied here? `#[serde(skip_serializing_if = "Option::is_none")]` ?

Comment (mhseiden) [reply src/header.rs]: Fwiw we've needed to use that attribute in a lot of our GCP + serde code...

Comment (Keats) [reply src/header.rs]: @bhendo yes it should, I just pushed  a patch

**Comment by mhseiden** [comment]
The patch plays nice with our codebase; thanks for this!

**Comment by bhendo** [comment]
@Keats looks like these changes do the trick for me. Thanks!

**Comment by Keats** [comment]
@naerbnic ping if you want to have a look before a I merge it

**Pull request merged**